File: py/1822.py
class Solution(object):
    def arraySign(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        p = 1
        for num in nums:
            p *= num
        
        if p > 0:
            return 1
        elif p == 0:
            return 0
        else:
            return -1

File: java/1822.java
class Solution {
    public int arraySign(int[] nums) {
        int p = 1;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] > 0) {
                p *= 1;
            }
            else if (nums[i] == 0) {
                return 0;
            }
            else {
                p *= -1;
            }
        }
        return p;
    }
}